causalml/inference/tree/models.py

```diff
@@ -1008,47 +904,36 @@ def growDecisionTreeFrom(self, X, treatment, y, evaluationFunction, max_depth=10
                                                           parentNodeSummary=currentNodeSummary)
 
                 # check the split validity on min_samples_treatment
-                if set(leftNodeSummary.keys()) != set(rightNodeSummary.keys()):
-                    continue
-                node_mst = 10**8
-                for ti in leftNodeSummary:
-                    node_mst = np.min([node_mst, leftNodeSummary[ti][1]])
-                    node_mst = np.min([node_mst, rightNodeSummary[ti][1]])
+                assert len(leftNodeSummary) == len(rightNodeSummary)
+
+                node_mst = min([stat[1] for stat in leftNodeSummary + rightNodeSummary])
                 if node_mst < min_samples_treatment:
                     continue
-                # evaluate the split
 
-                if evaluationFunction == self.evaluate_CTS:
-                    leftScore1 = evaluationFunction(leftNodeSummary)
-                    rightScore2 = evaluationFunction(rightNodeSummary)
+                # evaluate the split
+                if self.evaluationFunction == self.evaluate_CTS:
+                    leftScore1 = self.evaluationFunction(leftNodeSummary)
+                    rightScore2 = self.evaluationFunction(rightNodeSummary)
                     gain = (currentScore - p * leftScore1 - (1 - p) * rightScore2)
                     gain_for_imp = (len(X) * currentScore - len(X_l) * leftScore1 - len(X_r) * rightScore2)
-                elif evaluationFunction == self.evaluate_DDP:
-                    if self.control_name in leftNodeSummary and self.control_name in rightNodeSummary:
-                        leftScore1 = evaluationFunction(leftNodeSummary, control_name=self.control_name)
-                        rightScore2 = evaluationFunction(rightNodeSummary, control_name=self.control_name)
-                        gain = np.abs(leftScore1 - rightScore2)
-                        gain_for_imp = np.abs(len(X_l) * leftScore1 - len(X_r) * rightScore2)
-                    else:
-                        gain = 0
+                elif self.evaluationFunction == self.evaluate_DDP:
+                    leftScore1 = self.evaluationFunction(leftNodeSummary, control_name=self.control_name)
```

Comment:
`control_name` parameter seems to be removed in `evaluate_DDP()`, we then need to change these two lines accordingly

Reply:
Done. Thanks for catching this!